File: MinimizingCoins.py
m = float("inf")


def getCoins(arr, n, x, dp):
    for i in range(2, n + 1):
        for j in range(1, x + 1):
            if j - arr[i - 1] >= 0:
                dp[i][j] = min(dp[i - 1][j], 1 + dp[i][j - arr[i - 1]])
            else:
                dp[i][j] = dp[i - 1][j]
    return dp[n][x]


n, x = map(int, input().split())
arr = [int(x) for x in input().split()]
dp = [[0 for _ in range(x + 1)] for _ in range(n + 1)]

for i in range(x + 1):
    dp[0][i] = m
for i in range(1, x + 1):
    if i % arr[0] == 0:
        dp[1][i] = i // arr[0]
    else:
        dp[1][i] = m
ans = getCoins(arr, n, x, dp)
if ans == m:
    print(-1)
else:
    print(ans)

File: MinimizingCoins.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int m = INT_MAX;
	int n, x;
	cin >> n >> x;
	int arr[n + 1];
	for(int i = 0; i < n; i ++){
	    cin >> arr[i];
	}
	vector<int> dp(x + 1, 1e9);
	dp[0] = 0;
	for(int i = 1; i <= x; i ++){
	    for (int j = 0;j < n; j++){
	        if(i - arr[j] >= 0){
	            dp[i] = min(dp[i], dp[i - arr[j]] + 1);
	        }
	    }
	}
	cout << (dp[x] == 1e9 ? -1 : dp[x]);
	
	
}